pkg/health/health_check_handler.go

```diff
@@ -36,8 +36,8 @@ func (srv *Service) getHealthInfo() map[string]interface{} {
 }
 
 // HealthCheckHandler returns a default heath check result.
-func (srv *Service) HealthCheckHandler(ctx *gin.Context) {
-	// default handler for system health
+func (srv *HealthCheckService) GetHealthCheckHandler(ctx *gin.Context) {
```

Comment:
Yes, i was thinking as what's done in Java. But furthermore, how would this look if we had to handle more than just a Get for a service? For instance, if in signup we have both a Get and Post how would the handle functions be named? I think there needs to be a way that is indicative of what http action the handler is handling...

Reply:
i think it is also common practice even in golang to use Get, Create, Delete for handler function names.. see these examples: 

1. https://github.com/sarulabs/di-example/blob/master/app/handlers/cars.go
2. https://itnext.io/structuring-a-production-grade-rest-api-in-golang-c0229b3feedc
3. https://medium.com/@adigunhammedolalekan/build-and-deploy-a-secure-rest-api-with-go-postgresql-jwt-and-gorm-6fadf3da505b

Reply:
Oh... this `Get` is for `GET` http method? Then yeah.. having it in the name could be good solution. Just not sure if we should add it in the beginning.. We can easily confuse it with getters/settes as I just did :) Maybe `HealthCheckGetHandler`?
Or add Get/Post/etc only when we need to have multiple methods like for `GetSignupHandler` and `PostSignupHandler`. But I'm fine with keeping it `GetHealtCheckHandler` too.

@sbryzak WDYT?

Reply:
> i think it is also common practice even in golang to use Get, Create, Delete for handler function names.. see these examples:

Yep. It looks like it is. +1 to have it as `GetXXXHandler`, `PostXXXHandler`, etc.

Reply:
I would avoid adding `Get` here also.  If for example we modified the endpoint to instead use `POST`, it would mean we would also need to rename this function.

Reply:
Just to add to this, @tinakurian all of your examples are of REST services but the registration service isn't REST.  If we *were* developing a REST service, I would approach this differently.

Reply:
@sbryzak i'm not sure i understand - this is definitely rest.